### PR TITLE
feat: add post notification api

### PR DESCRIPTION
**Pull request by mabdh**
- add post notification api
- deprecate notify receiver api
- update flow to post alerts to be reusable by event notification

### CODE REVIEW

File: internal/store/postgres/migrations/000016_replace_idempotencies_success_notification_id_column.down.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+ALTER TABLE
+  notifications
+DROP COLUMN IF EXISTS receiver_selectors;
+
```
Comment (manishdangi98) [inline internal/store/postgres/migrations/000016_replace_idempotencies_success_notification_id_column.down.sql:4]: @mabdh why we are dropping -> adding and again dropping the receiver_selectors column.

Comment (mabdh) [reply internal/store/postgres/migrations/000016_replace_idempotencies_success_notification_id_column.down.sql:4]: It is up and down. Am I missing something there?

ref
https://github.com/golang-migrate/migrate

Comment (ishanarya0) [reply internal/store/postgres/migrations/000016_replace_idempotencies_success_notification_id_column.down.sql:4]: @manishdangi98 there are the up and down queries. The up file is used to implement the changes and down file is used for rollback. Down is the reverse if Up.

**Review by manishdangi98 — APPROVED** [review]
LGTM

**Pull request merged**